### PR TITLE
New Posts / Pages status and author filter bar

### LINKED ISSUE #8823: Update UI for Posts / Pages lists status and author filters
The dropdown post status filter that currently resides in the navigation bar of the posts and pages list view controllers has some discoverability issues, and is a little clunky.

Let's replace it with a scrollable tab view, which makes it more obvious that other filters are available. Designs from @sylvesterwilmott:

aa5cf6a6-e0f8-4492-ab56-d7dd05f8347b

641d91a5-3651-4379-aee1-45e4c45d8a70

d2d114a6-198e-4471-bec5-007bcabe526e

### PR DESCRIPTION
**Pull request by frosty**
Fixes #8823. This PR replaces the post status dropdown filter that was in the navigation bar of the Posts and Pages screens, as well as the author segmented control that was in the Posts screen. 

The most recent reference designs can be found here: https://github.com/wordpress-mobile/WordPress-iOS/issues/8823#issuecomment-372324567. I'm currently referencing a Gridicons branch for the new multiple users icon, but I'll update this PR once that's merged.

filter-tabs

Note that self-hosted (non-Jetpack) sites will currently show a placeholder Gravatar.

Apologies for the large PR!

**To test:**

* For both Posts and Pages, ensure that the filter bar displays correctly.
* Ensure that you can switch between the tabs, and that the content is updated as expected.
* Check that if you leave the posts / pages screen and then return, your selection is restored.
* Check that the navigation bar dropdown is no longer presenter.
* Check that the search bar still works as expected.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/AuthorFilterButton.swift [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        static let contentSize = CGSize(width: 72.0, height: 44.0)
+        static let gravatarSize = CGSize(width: 28.0, height: 28.0)
+        static let multipleUsersGravatarSize = CGSize(width: 20.0, height: 20.0)
+        static let stackViewSpacing: CGFloat = 7.0
+        static let chevronVerticalPadding: CGFloat = 2.0
+        static let leadingPadding: CGFloat = 12.0
+    }
+}
+
+fileprivate extension UIImage {
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/AuthorFilterButton.swift]: Note on this one! `fileprivate` (AFAIK) in this case behaves exactly as if it was `private`. Accessible by everyone!

File: WordPress/Classes/ViewRelated/Post/AuthorFilterViewController.swift [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+    private let rows = [
+        AuthorFilter.mine,
+        AuthorFilter.everyone
+    ]
+
+    init(initialSelection: PostListFilterSettings.AuthorFilter,
+         gravatarEmail: String? = nil,
+         onSelectionChanged: ((PostListFilterSettings.AuthorFilter) -> Void)? = nil) {
+        self.gravatarEmail = gravatarEmail
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/AuthorFilterViewController.swift]: Nipticky: can we add a newline between the last signature line, and the first line that actually belongs to the body? (it's a bit hard to grasp, at a glance, where one begins and the other ends)

File: WordPress/Classes/ViewRelated/Post/AuthorFilterViewController.swift [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        titleLabel.font = Fonts.titleFont
+        return titleLabel
+    }()
+
+    private let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.spacing = 10.0
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/AuthorFilterViewController.swift]: Maybe another `Metric` constant here?

Comment (frosty) [reply WordPress/Classes/ViewRelated/Post/AuthorFilterViewController.swift]: Good catch :)

File: WordPress/Classes/ViewRelated/Post/AuthorFilterButton.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Post/AuthorFilterButton.swift]: Whoops, not sure why I did that! Thanks.

**Comment by iamthomasbishop** [comment]
One minor minor thing - and maybe it's just the coloring of the gif -  should we darken the multi-user icon in the filter bar? It looks like it's gray or gray-lighten-10, maybe should be slightly darker. // cc @SylvesterWilmott for final say

**Comment by frosty** [comment]
@iamthomasbishop Thanks for raising this – I missed that it's actually darker in the designs:

screen shot 2018-03-13 at 14 42 09

I'll push a fix.

**Comment by frosty** [comment]
@jleandroperez Ready for another look! Thanks for the great review! We had some discussions about the transition animation – for now, I've improved the look of it slightly but we may post a future change to move the filter bar below the search bar.

**Comment by jleandroperez** [comment]
@frosty looking great!! minor note though: verified (all of the) fixes in `Posts`, but not on `Pages`. Is that expected?. By fixes i mean:

- Improved UISearchDisplayController animation
- Selected filter not showing up below the search bar

Thank you!!!

**Comment by frosty** [comment]
@jleandroperez Whoops. Sorry I missed that! I pushed the same changes for the Pages screen as well. Hopefully that's everything!

**Pull request merged**